File: Trashare/DEtailViewController.m
//
//  DetailViewController.m
//  Trashare
//
//  Created by Marina Huber on 11/6/15.
//  Copyright © 2015 The App Academy. All rights reserved.
//

#import "DetailViewController.h"
#import "Trashare-Swift.h"
#import <UIKit/UIKit.h>

@interface DetailViewController () 

@end

@implementation DetailViewController




- (void)viewDidLoad
{
    [super viewDidLoad];

    self.titleTrash.text = self.descriptionString;
   
    self.dateTrash.text = self.dateCreated;

}

- (IBAction)cancelButton:(id)sender
{
    [self.navigationController popViewControllerAnimated:YES];
}

- (void) viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:YES];
	[self.navigationController setNavigationBarHidden:YES animated:YES];
	self.view.backgroundColor = [UIColor blackColor];
    
}

- (IBAction)tap:(id)sender {
    
        if (!isFullScreen) {
            [UIView animateWithDuration:0.4 delay:0 options:0 animations:^{
                prevFrame = self.showImage.frame;
                [self.showImage setFrame:[[UIScreen mainScreen] bounds]];
            }completion:^(BOOL finished){
                isFullScreen = YES;
            }];
            return;
        }
        else{
            self.titleTrash.alpha = 1.0;
            self.dateTrash.alpha = 1.0;

            [self.navigationController setNavigationBarHidden:NO animated:YES];
            [UIView animateWithDuration:4 delay:0 options:0 animations:^{
                [self.showImage setFrame:prevFrame];
            }completion:^(BOOL finished){
                isFullScreen = NO;
            }];
            return;
        }
    
}

@end

File: Trashare/ViewController/DetailViewController.swift
//
//  DetailViewController.swift
//  Trashare
//
//  Created by Marina Huber on 4/18/19.
//  Copyright © 2019 The App Academy. All rights reserved.
//

import UIKit
// TODO: Find localized string solution for all the strings for Parse

@objc class DetailViewController: UIViewController, UINavigationControllerDelegate, UIScrollViewDelegate {
	@IBOutlet weak var gradientView: UIView!
	@IBOutlet private var dateTrash: UILabel!
	@objc var currentObject : PFObject?

	// PFFileObject representes a file of binary data stored on the Parse servers.
	@IBOutlet var showImage: UIImageView!
	@IBOutlet private var titleTrash: UILabel!

	override func viewDidLoad() {
		super.viewDidLoad()
		setGradientBackground()
		if let object = currentObject {
			self.titleTrash.text = object["titleTrashare"] as? String
		if let fileImage = object["imageFile"] as? PFFileObject {
			getImageFromData(data: fileImage)
			}
		}
	}

	func getImageFromData(data: PFFileObject) {
		data.getDataInBackground {(data: Data?, error: Error?) in
			if let data = data, let image = UIImage(data: data) {
				self.showImage.image = image
			} else {}
		}
	}


	func setGradientBackground() {
		let colorTop =  UIColor(red: 209.0/255.0, green: 255.0/255.0, blue: 100.0/255.0, alpha: 1.0).cgColor
		let colorBottom = UIColor(red: 174.0/255.0, green: 255.0/255.0, blue: 158.0/255.0, alpha: 1.0).cgColor

		let gradientLayer = CAGradientLayer()
		gradientLayer.colors = [colorTop, colorBottom]
		gradientLayer.locations = [0.7, 1.0]
		gradientLayer.frame = self.view.bounds

		self.gradientView.layer.insertSublayer(gradientLayer, at: 0)
	}

	@IBAction func cancel(_ sender: Any) {
		navigationController?.popViewController(animated: true)
	}


}
